src/crate/qa/tests.py

```diff
@@ -188,13 +188,34 @@ def _new_cluster(self, version, num_nodes, settings=None, env=None):
         nodes = []
         for id in range(num_nodes):
             s['node.name'] = s['cluster.name'] + '-' + str(id)
-            nodes.append(self._new_node(version, s, env)[0])
+            nodes.append(self._new_node(version, crate_root, s, env)[0])
+        return CrateCluster(nodes)
+
+    def _new_heterogeneous_cluster(self, available_versions, num_nodes, crate_root=None, settings=None, env=None):
```

Comment:
`crate_root` and `env` are never used. Can we also remove the params here?

Reply:
And maybe rename `available_versions` to `versions` or so. Not sure what `available` adds here. Having a `unavailable_versions` or something like that wouldn't make sense.